triqs/arrays/mpi.hpp

```diff
@@ -70,8 +70,13 @@ namespace arrays {
   template <typename A> REQUIRES_IS_ARRAY mpi_broadcast(A &a, mpi::communicator c = {}, int root = 0) {
    if (!has_contiguous_data(a)) TRIQS_RUNTIME_ERROR << "Non contiguous view in mpi_broadcast";
    auto sh = a.shape();
+   auto m_pos = a.indexmap().memory_layout().get_memory_positions();
    MPI_Bcast(&sh[0], sh.size(), mpi::mpi_datatype<typename decltype(sh)::value_type>(), root, c.get());
-   if (c.rank() != root) resize_or_check_if_view(a, sh);
+   MPI_Bcast(&m_pos[0], m_pos.size(), mpi::mpi_datatype<typename decltype(m_pos)::value_type>(), root, c.get());
+   if (c.rank() != root) {
+     resize_or_check_if_view(a, sh);
+     set_mem_layout_or_check_if_view(a, memory_layout_t<A::rank>(m_pos));
```

Comment:
This function should NEVER be invoked without a proper resize, am I correct?
Maybe we could just wrap the two operations into one function.

Reply:
Do you want to call `resize_or_check_if_view()` from `set_mem_layout_or_check_if_view()` or rather call both of them from yet another function? Can you suggest a name for the combined function?